core/teca_threaded_algorithm.cxx

```diff
@@ -201,18 +210,49 @@ const_p_teca_dataset teca_threaded_algorithm::request_data(
                 }
             }
 
-            // get the requested data. will block until it's ready.
-            this->internals->thread_pool->wait_data(input_data);
-            )
-
-        // execute override
-        TECA_PROFILE_PIPELINE(128, alg, "execute", port,
-            out_data = alg->execute(port, input_data, request);
-            )
+            int n_tasks_remaining = 0;
+            do
+            {
+               // get the requested data. will block until it's ready.
+               std::vector<const_p_teca_dataset> input_data;
+
+               n_tasks_remaining = this->internals->thread_pool->wait_some(
+                    this->stream_size, this->poll_interval, input_data);
+
+                // when stremaing recycle last round's output, this is
```

Comment:
stremaing -> streaming

Reply:
found this misspelling in 3 other places. fixed